evennia/commands/default/building.py

```diff
@@ -599,22 +595,26 @@ def func(self):
             string = ""
             name = objdef["name"]
             aliases = objdef["aliases"]
-            typeclass = objdef["option"]
 
-            # create object (if not a valid typeclass, the default
-            # object typeclass will automatically be used)
-            lockstring = self.new_obj_lockstring.format(id=caller.id)
-            obj = create.create_object(
-                typeclass,
+            obj_typeclass, errors = caller.get_object_typeclass(obj_type="object", typeclass=objdef["option"])
+            if errors:
+                self.msg(errors)
+            if not obj_typeclass:
+                continue
+
+            obj, errors = obj_typeclass.create(
                 name,
                 caller,
                 home=caller,
                 aliases=aliases,
-                locks=lockstring,
                 report_to=caller,
+                creator=caller
```

Comment:
It seems suitable to have a method of 'cmd_create` here. There are many reasons to call this creation method.